src/backend/distributed/commands/multi_copy.c

```diff
@@ -1751,6 +1752,12 @@ CitusCopyDestReceiverStartup(DestReceiver *dest, int operation,
 	LockShardListMetadata(shardIntervalList, ShareLock);
 	LockShardListResources(shardIntervalList, ShareLock);
 
+	/*
+	 * We might have some concurrent metadata changes and we need to accept the cache
+	 * invalidation messages to learn about the changes.
+	 */
+	AcceptInvalidationMessages();
```

Comment:
I wonder whether it makes sense to always do `AcceptInvalidationMessages()` before checking whether a cache entry is valid in `metadata_cache.c`.